packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -669,9 +669,14 @@ abstract class SchedulerBinding extends BindingBase with ServicesBinding {
   /// If a frame has already been scheduled with [scheduleFrame] or
   /// [scheduleForcedFrame], this call may delay that frame.
   ///
+  /// If any scheduled frame has already begun or if another
+  /// [scheduleWarmUpFrame] was already called, this call will be ignored.
+  ///
   /// Prefer [scheduleFrame] to update the display in normal operation.
   void scheduleWarmUpFrame() {
-    assert(!_warmUpFrame);
+    if (_warmUpFrame && !(schedulerPhase == SchedulerPhase.idle))
```

Comment:
prefer `!=` over `!( == )`

Reply:
also i'm pretty sure you want `||` not `&&` :-)

Reply:
T.T